File: yandex_algo/linked-list_A.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"slices"
	"strconv"
	"strings"
)

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    scanner.Scan()
    q, _ := strconv.Atoi(scanner.Text())
    
    var lst []int
    lineNum := 0
    
    for scanner.Scan() && lineNum < q {
        parts := strings.Fields(scanner.Text())
        lineNum++
        
        typ, _ := strconv.Atoi(parts[0])
        
        switch typ {
        case 1:
            x, _ := strconv.Atoi(parts[1])
            y, _ := strconv.Atoi(parts[2])
            
            if x == 0 {
                // Вставка в начало
                lst = append([]int{y}, lst...)
            } else {
                // Вставка после x-го элемента
                lst = slices.Insert(lst, x+1, y)
            }
            
        case 2:
            x, _ := strconv.Atoi(parts[1])
            fmt.Println(lst[x-1])
            
        case 3:
            x, _ := strconv.Atoi(parts[1])
            // Удаление x-го элемента
            lst = append(lst[:x-1], lst[x:]...)
        }
    }
}

File: yandex_algo/linked-list_A.py
import sys

# input = sys.stdin.read
# data = input().split()

data = [
    "8",
    "1",
    "0",
    "5",
    "2",
    "1",
    "3",
    "1",
    "1",
    "0",
    "6",
    "1",
    "0",
    "7",
    "2",
    "2",
    "1",
    "1",
    "5",
    "2",
    "2",
]

q = int(data[0])
index = 1

# Изначально пустой список
lst: list = []

for i in range(q):
    typ = int(data[index])
    index += 1

    if typ == 1:
        # Запрос 1: добавить y после x-го элемента (x=0 -> в начало)
        x = int(data[index])
        y = int(data[index + 1])
        index += 2

        if x == 0:
            lst.insert(0, y)
        else:
            lst.insert(x + 1, y)

    elif typ == 2:
        # Запрос 2: вывести элемент на позиции x
        x = int(data[index])
        index += 1
        print(lst[x - 1])  # Индексация с 0, позиции с 1

    elif typ == 3:
        # Запрос 3: удалить элемент на позиции x
        x = int(data[index])
        index += 1
        del lst[x - 1]
